testing/matrix/linter/rules/resources/vpa.go

```diff
@@ -121,6 +130,61 @@ func fillVPAMaps(scope *lintingScope, vpaTargets map[storage.ResourceIndex]struc
 	if label, ok := labels["workload-resource-policy.deckhouse.io"]; ok {
 		vpaTolerationGroups[target] = label
 	}
+
+	vnm, ok := parseVPAResourcePolicyContainers(scope, vpa)
+	if !ok {
+		return
+	}
+	vpaContainerNamesMap[target] = vnm
+}
+
+// parseVPAResourcePolicyContainers parses VPA containers names in ResourcePolicy and check if minAllowed and maxAllowed for container is set
+func parseVPAResourcePolicyContainers(scope *lintingScope, vpaObject storage.StoreObject) (map[string]struct{}, bool) {
+	var containers []string
+	res := make(map[string]struct{})
```

Comment:
In go_lib, we have a Set datastructure that makes handling strings more convenient. We use is in several hooks, in Upmeter, and probably in other places. It has some handy methods. Take a look.

```go

import (
  ...
  "github.com/deckhouse/deckhouse/go_lib/set"
)

...



containerts := set.New()

for ... := range containers {
  containerName := ...

  containers.Add(containerName)
}

```

Reply:
Good point, thanks !